### ISSUE TITLE
XLA operation semantics documentation BatchNormTrain error

### ISSUE THREAD

**Issue by mateoespinosa** [tensorflow/tensorflow#14676]
Hey! I think there are a few errors in the documentation for the XLA BatchNormTrain operation in https://www.tensorflow.org/performance/xla/operation_semantics#batchnormgrad.

The gradient of the scaling factor `gamma` should be
<a href="https://www.codecogs.com/eqnedit.php?latex=\nabla&space;\gamma&space;=&space;\text{sum}\Big(&space;\nabla&space;y&space;*&space;\frac{(&space;x&space;-&space;\mu&space;)}{&space;\sqrt{\sigma^2&space;&plus;&space;\epsilon}}&space;\Big)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\nabla&space;\gamma&space;=&space;\text{sum}\Big(&space;\nabla&space;y&space;*&space;\frac{(&space;x&space;-&space;\mu&space;)}{&space;\sqrt{\sigma^2&space;&plus;&space;\epsilon}}&space;\Big)" title="\nabla \gamma = \text{sum}\Big( \nabla y * \frac{( x - \mu )}{ \sqrt{\sigma^2 + \epsilon}} \Big)" /></a>
instead of 
<a href="https://www.codecogs.com/eqnedit.php?latex=\nabla&space;\gamma&space;=&space;\text{sum}\Big(&space;\nabla&space;y&space;*&space;(&space;x&space;-&space;\mu&space;)&space;*&space;\sqrt{\sigma^2&space;&plus;&space;\epsilon}&space;\Big)" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\nabla&space;\gamma&space;=&space;\text{sum}\Big(&space;\nabla&space;y&space;*&space;(&space;x&space;-&space;\mu&space;)&space;*&space;\sqrt{\sigma^2&space;&plus;&space;\epsilon}&space;\Big)" title="\nabla \gamma = \text{sum}\Big( \nabla y * ( x - \mu ) * \sqrt{\sigma^2 + \epsilon} \Big)" /></a>

Moreover, the gradient for the input tensor is also not correct. It should instead read something like this:
<a href="https://www.codecogs.com/eqnedit.php?latex=\Big(&space;\nabla&space;y&space;-&space;\frac{\hat{x}&space;\times&space;\nabla&space;\gamma&space;&plus;&space;\nabla&space;\beta}{m&space;w&space;h}\Big)&space;\times&space;\frac{\gamma}{\sqrt{\sigma^2&space;&plus;&space;\epsilon}}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\Big(&space;\nabla&space;y&space;-&space;\frac{\hat{x}&space;\times&space;\nabla&space;\gamma&space;&plus;&space;\nabla&space;\beta}{m&space;w&space;h}\Big)&space;\times&space;\frac{\gamma}{\sqrt{\sigma^2&space;&plus;&space;\epsilon}}" title="\Big( \nabla y - \frac{\hat{x} \times \nabla \gamma + \nabla \beta}{m w h}\Big) \times \frac{\gamma}{\sqrt{\sigma^2 + \epsilon}}" /></a>

Where the products and summations are done in a "broadcasted" manner when shapes don't match and <a href="https://www.codecogs.com/eqnedit.php?latex=\hat{x}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\hat{x}" title="\hat{x}" /></a> is the whitened input tensor. That is:
<a href="https://www.codecogs.com/eqnedit.php?latex=\hat{x}&space;=&space;\frac{x&space;-&space;\mu}{\sqrt{\sigma^2&space;&plus;&space;\epsilon}}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\hat{x}&space;=&space;\frac{x&space;-&space;\mu}{\sqrt{\sigma^2&space;&plus;&space;\epsilon}}" title="\hat{x} = \frac{x - \mu}{\sqrt{\sigma^2 + \epsilon}}" /></a>

There also seem to be a typo in the table explaining the output of this XLA node. The second row should say `grad_offset` in the first column, `ComputationalDataHandle` in the second column, and the semantics column should say something like `gradient with respect to input offset`.

 
Finally, it may be good if it could be specified that the summation used for the gradients of the scaling and offset tensors is performed over all dimensions that were used to compute the different statistics during normalization.

**Comment by yunxing** [comment]
Just saw this. Good catches, thanks! We did notice the mistakes in the formula when we were testing the implementation, but haven't got a chance to fix the doc. Will work on a fix.

**Comment by mateoespinosa** [comment]
Great! Let me know if I can help somehow. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yunxing: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yunxing: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yunxing** [comment]
Ahh this is has already been fixed a while ago, I forgot to close this.
